### PR TITLE
Docs: add code reference

### LINKED ISSUE #469: Include a code reference which allows users to search functions of autora and its optional dependencies.

### PR DESCRIPTION
**Pull request by hollandjg**
# Description

Include a working code reference again.

*If the issue is on Github, simply link to it using the '#' symbol; otherwise, provide a notion page link)*:
- resolves #469 

# Type of change

- **docs**: Documentation only changes

# Features (Optional)
- use additional multi repo imports to generate a working code reference
- include these with a literate-nav hierarchy
- refactor: hooks.py – replace deprecated mkdocs-simple-hooks with mkdocs' built in hooks

### CODE REVIEW

**Comment by hollandjg** [comment]
Thanks @younesStrittmatter and @benwandrew for the feedback!

**Review by younesStrittmatter — APPROVED** [review]
Perfect!

**Review by benwandrew — APPROVED** [review]
hard won!

**Comment by hollandjg** [comment]
Hi @younesStrittmatter and @benwandrew – one last change – I've made the titles of the pages more descriptive. What do you think?

**Review by younesStrittmatter — APPROVED** [review]
Very good! Love the new titles!

File: mkdocs/generate_code_reference.py [hunk 1]
```diff
@@ -29,12 +29,55 @@ def source_file_generator(src_paths):
...
+            # a.b
+            <BLANKLINE>
+            ::: a.b
+
+            >>> print(get_reference_file_string(("a", "b", "__init__")))
+            # a.b
+            <BLANKLINE>
+            ::: a.b.__init__
+
+                        >>> print(get_reference_file_string(("a", "b")))
```
Comment (benwandrew) [inline mkdocs/generate_code_reference.py]: for my own instruction, what is this last example adding that isn't in the one in lines 53-56 above?. i see the extra indentation, but i don't understand what is incremental. thanks!

**Review by benwandrew — APPROVED** [review]
excellent! just a question for my own edification

File: mkdocs/generate_code_reference.py (hunk 1 above)
Comment (hollandjg) [reply mkdocs/generate_code_reference.py]: It's broken! Thanks for catching that, I'll take it out.

**Pull request merged**